keras_cv/src/models/object_detection/faster_rcnn/faster_rcnn.py

```diff
@@ -0,0 +1,477 @@
+        # Begin construction of forward pass
+        image_shape =  feature_extractor.input_shape[1:]
+        if None in image_shape:
+            raise ValueError("Image shape should not have None")
+        
+        images = keras.layers.Input(
+           image_shape, 
+           name="images",
+        )
+        backbone_outputs = feature_extractor(images)
+        feature_map = feature_pyramid(backbone_outputs)
+        
+        
+        # [BS, num_anchors, 4], [BS, num_anchors, 1]
+        rpn_boxes, rpn_scores = rpn_head(feature_map)
+
+        for lvl in rpn_boxes:
+            rpn_boxes[lvl] = keras.layers.Reshape(
+                target_shape=(-1, 4))(rpn_boxes[lvl])
+        
+        for lvl in rpn_scores:
+            rpn_scores[lvl] = keras.layers.Reshape(
+                target_shape=(-1, 1))(rpn_scores[lvl])
+        
+        rpn_cls_pred = keras.layers.Concatenate(axis=1, name="rpn_classification")(
+            tree.flatten(rpn_scores)
+        )
+        rpn_box_pred = keras.layers.Concatenate(axis=1, name="rpn_box")(
+            tree.flatten(rpn_boxes)
+        )
+        
+        anchors = anchor_generator(image_shape=image_shape)
+        decoded_rpn_boxes = _decode_deltas_to_boxes(
+            anchors=anchors,
+            boxes_delta=rpn_boxes,
+            anchor_format=bounding_box_format,
+            box_format=bounding_box_format,
+            variance=BOX_VARIANCE,
+        )
+        # Generate ROI's from RPN head
+        rois, _ = roi_generator(decoded_rpn_boxes, rpn_scores)
+        
+        feature_map = roi_pooler(features=feature_map, boxes=rois)
+        
+        # Reshape the feature map [BS, H*W*K]
+        feature_map = keras.layers.Reshape(
+            target_shape=(rois.shape[1],  -1))(feature_map)
+        
+        # Pass final feature map to RCNN Head for predictions
+        box_pred, cls_pred = rcnn_head(feature_map=feature_map)
+        
+        box_pred = keras.layers.Concatenate(axis=1, name="box")([box_pred])
+        cls_pred = keras.layers.Concatenate(axis=1, name="classification")(
+            [cls_pred]
+        )
+        
+        inputs = {"images": images}
+        outputs = {
+            "box": box_pred,
+            "classification": cls_pred,
+            "rpn_box": rpn_box_pred,
+            "rpn_classification": rpn_cls_pred,
+        }
+        
+        super().__init__(
+            inputs=inputs,
+            outputs=outputs,
+            **kwargs,
+        )
+        
+        # Define the model parameters
+        self.bounding_box_format = bounding_box_format
+        self.anchor_generator = anchor_generator
+        self.num_classes = num_classes
+        self.rpn_labeler = label_encoder or _RpnLabelEncoder(
+            anchor_format=bounding_box_format,
+            ground_truth_box_format=bounding_box_format,
+            positive_threshold=0.7,
+            negative_threshold=0.3,
+            samples_per_image=256,
+            positive_fraction=0.5,
```

Comment:
These args are also good candidates for input parameters of this class.

Reply:
Here user can pass custom `label_encoder` or else by default these are the args we consider.